src/coreclr/jit/emitarm64.cpp

```diff
@@ -6906,7 +6924,58 @@ void emitter::emitIns_R_R_R_I(instruction ins,
     }
     assert(fmt != IF_NONE);
 
-    instrDesc* id = emitNewInstrCns(attr, imm);
+    // An "instrDesc" will *always* be required.
+    // Under normal circumstances the instruction
+    // will be added to the emitted group. However,
+    // this is not correct for instructions that
+    // are going to overwrite already-emitted
+    // instructions and we therefore need space to
+    // hold the new instruction descriptor.
+    instrDesc* id;
```

Comment:
Since this can happen only for `ldp` and `stp`, this block can be inside the `if (isLdSt)` check above.

Reply:
Following a change called for by Jakob Botsch, the code has been changed to build the instruction directly on top of the previously emitted instruction. If you are still unhappy with the structure of the changed code, please advise and I will revisit.

Reply:
We should still move this code inside `isLdSt` because at least currently, that's the only scenario we are using it for. Here you can have something like:

```c++
instrDesc* id = nullptr;

// moved logic to populate `id`.
...
...
// somewhere around line 6934
if (reuseInstr == nullptr)
{
  id = emitNewInstrCns(attr, imm);
}
else
{
  assert(id != nullptr);
}

Also, have you considered @BruceForstall proposal in https://github.com/dotnet/runtime/pull/77540#issuecomment-1299600281?

Reply:
Bruce's proposal has been implemented, so reuseInstr is no longer a variable.